### PR TITLE
helm: use IfNotPresent for versioned helm chart release

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/envoyproxy/gateway/issues/3183

### CODE REVIEW

**Review by guydc — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

File: test/helm/default.yaml [hunk 1]
```diff
@@ -369,7 +369,7 @@ spec:
         - name: KUBERNETES_CLUSTER_DOMAIN
           value: cluster.local
         image: docker.io/envoyproxy/gateway-dev:latest
-        imagePullPolicy: Always
+        imagePullPolicy: IfNotPresent
```
Comment (arkodg) [inline test/helm/default.yaml:372]: curious how it defaults to `IfNotPresent` when the default is `Always` in the Makefile
https://github.com/envoyproxy/gateway/blob/9a626f262317b90946219d7f85069221cce2c4b6/tools/make/kube.mk#L65

Comment (zirain) [reply test/helm/default.yaml:372]: is this used for helm install/template?

**Pull request merged**